src/core/constants/loading-instructions.js

```diff
@@ -5,30 +5,30 @@
  * Only `UNLOAD` is not defined by the "Lazy loading" spec at this time. It's
  * added here because it supersedes all other loading instructions in AMP.
  *
- * @enum {string}
+ * @enum {"auto" | "lazy" | "eager" | "unload"}
  */
 export const Loading_Enum = {
   /**
    * If parent is available, fallback to its loading strategy (e.g. based on
    * whether the document is visible or not).
    * If parent is not available, proceed with loading at your own discretion.
    */
-  AUTO: 'auto',
+  AUTO: /** @type {"auto"} */ ('auto'),
```

Comment:
Are you sure these annotations are necessary?

Reply:
Unfortunately yes. I also don't know if they have a `ts` equivalent (re. enum). We may want to change their representation a bit in that world.